components/match2/wikis/halo/match_group_input_custom.lua

```diff
@@ -62,28 +69,29 @@ function CustomMatchGroupInput.processMap(map)
 	return map
 end
 
-function CustomMatchGroupInput.processOpponent(record, date)
+function CustomMatchGroupInput.processOpponent(record, date, opponentIndex)
 	local opponent = Opponent.readOpponentArgs(record)
 		or Opponent.blank()
 
-	-- Convert byes to literals
-	if opponent.type == Opponent.team and opponent.template:lower() == 'bye' then
-		opponent = {type = Opponent.literal, name = 'BYE'}
-	end
+	-- Retrieve icon for teams and do tbd checks
+	if opponent.type == Opponent.team then
+		if opponent.template:lower() == MATCH_BYE then
+			opponent = {type = Opponent.literal, name = MATCH_BYE}
+		else
+			opponent.icon = opponentFunctions.getIcon(opponent.template)
+		end
+	elseif opponent.type ~= Opponent.literal then
+		if not ALLOWED_OPPONENT_TYPES[opponent.type or ''] then
+			error('Unsupported opponent type "' .. (opponent.type or '') .. '"')
+		end
+		opponent.match2players = matchFunctions.getPlayers(record, opponent.type, opponentIndex)
+		if Array.any(opponent.match2players, CustomMatchGroupInput._playerIsBye) then
+			opponent = {type = Opponent.literal, name = MATCH_BYE}
+		end
```

Comment:
Undo the changes here

Reply:
As in remove the added code and add back the removed code?

Reply:
Yes